fluent-bit-to-loki/pkg/client/client.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/gardener/logging/fluent-bit-to-loki/pkg/buffer"
+	"github.com/gardener/logging/fluent-bit-to-loki/pkg/config"
+	"github.com/grafana/loki/pkg/promtail/client"
+	"github.com/prometheus/common/model"
+)
+
+type newClientFunc func(cfg client.Config, logger log.Logger) (client.Client, error)
+
+// NewClient creates a new client based on the fluentbit configuration.
+func NewClient(cfg *config.Config, logger log.Logger) (client.Client, error) {
+	var ncf newClientFunc
+
+	if cfg.ReplaceOutOfOrderTS {
+		ncf = newTimestampOrderingClient
+	} else {
+		ncf = client.New
+	}
+
+	if cfg.BufferConfig.Buffer {
+		return buffer.NewBuffer(cfg, logger, ncf)
+	}
+	return ncf(cfg.ClientConfig, logger)
+}
+
+type clientWrapper struct {
+	lokiclient client.Client
+	lastTS     map[model.Fingerprint]time.Time
+	logger     log.Logger
+}
+
+// New makes a new dque loki client
```

Comment:
Does not match the func name. I also doubt that the description is not correct. There is nothing mentioned for `timestamp ordering` which is in the function name.